lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
It does. Just try the example. It sets `@bypass_confirmation_postpone` to true and doesn't reset it after save. Why save must be called in callbacks, maybe it's just application specific demand.

You don't check what field contains new email in https://github.com/plataformatec/devise/pull/4043/files#diff-b1fd8c2bf91721612bc850fce022cb14R404

New value should be in `email` but `assert_match "new_test@example.com", admin.email` fails, 'cause it's saved to `unconfirmed_email`. This is breaking change, but it has been released as a patch.

ping @ulissesalmeida 

I've run this test in 4.0.0 (made from this pr) and it passed:
```ruby
  test 'should send confirmation instructions by email after changing email from nil' do
    admin = create_admin(email: nil)
    # assert_email_sent "new_test@example.com" do
      assert admin.update_attributes(email: 'new_test@example.com')
    # end
    # assert_match "new_test@example.com", ActionMailer::Base.deliveries.last.body.encoded

    # added this line, it fails after pr:
    assert_match "new_test@example.com", admin.email
  end
```